lib/root_route_constraint.rb

```diff
@@ -0,0 +1,13 @@
+class RootRouteConstraint
+  def initialize
+    @current = Conference.where('end_date >= ?', Date.current).reorder(start_date: :asc)
+  end
+
+  def matches?(*)
+    if @current.present? && @current.first.splashpage.present?
+      @current.count == 1 && @current.first.splashpage.public?
```

Comment:
there needs to be a check for the presence of any conferences or the presence of splashpage or else `@current.first.splashpage.public?` give Nil:Class error. So if statement is used to check the presence.